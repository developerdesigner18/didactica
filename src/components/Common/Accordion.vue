<template>
  <div :class="['accordion-wrapper', variant]">
    <div
      v-for="(item, index) in items"
      :key="index"
      :class="['accordion-item', { active: isOpen(index) }]"
    >
      <div class="accordion-header" @click="toggle(index)">
        <img
          src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/arrow.svg"
          loading="lazy"
        />
        <span>{{ item.title }}</span>
      </div>
      <div
        v-if="isOpen(index)"
        class="accordion-content"
        v-html="item.content"
      ></div>
    </div>
  </div>
</template>
<script>
export default {
  name: "Accordion",
  props: {
    items: {
      type: Array,
      required: true
    },
    singleOpen: {
      type: Boolean,
      default: true
    },
    variant: {
      type: String,
      default: "variant1"
    }
  },
  data() {
    return {
      openIndexes: []
    };
  },
  mounted() {
    if (!this.singleOpen && this.items.length > 0) {
      this.openIndexes.push(0);
    }
  },
  methods: {
    toggle(index) {
      if (this.singleOpen) {
        this.openIndexes = this.isOpen(index) ? [] : [index];
      } else {
        if (this.isOpen(index)) {
          this.openIndexes = this.openIndexes.filter((i) => i !== index);
        } else {
          this.openIndexes.push(index);
        }
      }
    },
    isOpen(index) {
      return this.openIndexes.includes(index);
    }
  }
};
</script>
<style lang="scss" scoped>
.accordion-wrapper {
  .accordion-item {
    &:not(:first-child) {
      margin-top: 16px;
    }
    &.active {
      .accordion-header {
        img {
          transform: rotate(0deg);
        }
      }
    }
  }
  .accordion-header {
    display: flex;
    align-items: center;
    overflow: hidden;
    @include label-large($font-color: $brand-neutral-900, $font-weight: 500);
    cursor: pointer;
    span {
      margin-left: 8px;
    }
    img {
      transform: rotate(90deg);
    }
  }
  .accordion-content {
    padding-left: 24px;
    @include label-medium;
    color: $brand-neutral-900;
    letter-spacing: 0.4px;
  }
}
</style>
