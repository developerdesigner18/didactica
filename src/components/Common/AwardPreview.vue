<template>
  <div class="award__preview--wrapper">
    <div class="title" v-if="showTitle">{{ courseTitle }}</div>
    <div :class="['award-information', { 'mt-0': !showTitle }]">
      <div class="award-icon">
        <img :src="awardIcon" :alt="awardTitle" loading="lazy" />
      </div>
      <div
        :class="[
          'award-title',
          {
            'award-title__who':
              awardType !== 'course_participation_confirmation'
          }
        ]"
      >
        {{ awardTitle }}
      </div>
      <div class="presented-to">{{ presentedTo }}</div>
      <div class="user-name">{{ userName }}</div>
      <div class="award-description">
        {{ awardDescription }}
        <span>“{{ courseTitle }}”</span>
      </div>
      <div
        :class="[
          'issued-by',
          {
            'award-issued__by':
              awardType === 'course_participation_confirmation'
          }
        ]"
      >
        <div class="issued-text">
          <span>{{ issuedByText }}</span>
          <span v-html="issuedBy" />
        </div>
        <div class="logo" v-if="issuerLogo.length > 0">
          <div v-for="(l, i) in issuerLogo" :key="i">
            <img :src="l.logo" alt="logo" loading="lazy" />
          </div>
        </div>
      </div>
    </div>
    <div class="more-information">
      <div class="title">{{ moreInformationText }}</div>
      <div class="date-information">
        <div class="date-box issued-date">
          <div class="icon">
            <img
              src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/issued-on.svg"
              alt="issued on"
              loading="lazy"
            />
          </div>
          <div class="date">
            <span>{{ issuedOnText }}</span>
            <span>{{
              issuedOn !== "" ? formatDate(issuedOn) : "DD/MM/YYYY"
            }}</span>
          </div>
        </div>
        <div class="date-box expiring-date">
          <div class="icon">
            <img
              src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/expiring-on.svg"
              alt="expiration date"
              loading="lazy"
            />
          </div>
          <div class="date">
            <span>{{ expiringOn }}</span>
            <span>{{
              credential.expiration_date
                ? formatDate(credential.expiration_date)
                : "Does not expire"
            }}</span>
          </div>
        </div>
      </div>
      <div
        class="other-information"
        v-if="moreInformation && moreInformation.length > 0"
      >
        <Accordion :items="moreInformation" :singleOpen="false" />
      </div>
    </div>
    <div class="disclaimer">
      <div class="title">{{ disclaimerTitle }}</div>
      <div class="description">
        {{ disclaimer }}
      </div>
    </div>
  </div>
</template>

<script>
import { format } from "../../dateFormat";
import Accordion from "./Accordion.vue";
export default {
  components: {
    Accordion
  },
  props: {
    awardTitle: {
      type: String,
      required: true
    },
    courseTitle: {
      type: String,
      required: true
    },
    presentedTo: {
      type: String,
      default: "Presented to"
    },
    userName: { type: String, required: true },
    awardDescription: {
      type: String,
      default: "for the successful completion of the course"
    },
    issuedByText: {
      type: String,
      default: "Issued by"
    },
    moreInformationText: {
      type: String,
      default: "More Information"
    },
    issuedOnText: {
      type: String,
      default: "Issued On"
    },
    issuedOn: {
      type: String,
      default: ""
    },
    expiringOn: {
      type: String,
      default: "Expiring On"
    },
    disclaimerTitle: {
      type: String,
      default: "Disclaimer"
    },
    disclaimer: {
      type: String,
      default:
        "This award does not certify the learner as a trained health worker qualified to practice in this topic area. National and local certificates for qualification must be adhered to.  "
    },
    infoArr: {
      type: Array,
      required: true
    },
    award: {
      type: Object,
      required: true
    },
    showTitle: {
      type: Boolean,
      default: true
    }
  },
  computed: {
    awardType() {
      return this.award.content_type.slug;
    },
    credential() {
      return this.award.credential;
    },
    awardIcon() {
      let logo;
      if (this.awardType === "course_completion_award") {
        logo =
          "https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/award-of-completion.svg";
      } else if (this.awardType === "course_participation_confirmation") {
        logo =
          "https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/confirmation-of-participation.svg";
      }
      return logo;
    },
    issuedBy() {
      let text = "";
      if (this.awardType === "course_completion_award") {
        text = "WHO Academy";
      } else if (this.awardType === "course_participation_confirmation") {
        text = this.credential.issuers;
      }
      return text;
    },
    moreInformation() {
      let arr = [];
      this.infoArr.forEach((i) => {
        if (this.credential[i.key] !== "") {
          let obj = {
            title: i.title,
            content: this.credential[i.key]
          };
          arr.push(obj);
        }
      });
      return arr;
    },
    issuerLogo() {
      let logos = [];
      if (this.awardType === "course_participation_confirmation") {
        logos = this.credential.issuer_logos;
      } else {
        logos = [
          {
            logo: "https://whoalxppublicstorage.blob.core.windows.net/user-feedback/images/whoa-logo-black-en.svg"
          }
        ];
      }
      return logos;
    }
  },
  methods: {
    formatDate(date) {
      return format(date);
    }
  }
};
</script>
<style lang="scss" scoped>
.award__preview--wrapper {
  width: 100%;
  > .title {
    text-align: center;
    @include subtitle-large($brand-neutral-900);
    letter-spacing: 0.055px;
  }
  .award-information {
    margin-top: 28px;
    border-radius: 8px;
    background: linear-gradient(
      299deg,
      $brand-neutral-100 32.98%,
      $brand-neutral-0 84.25%
    );
    padding: 12px 16px 24px;
    @include flex-column-align-center;
    box-shadow: 0px 1px 3px 1px rgba(33, 33, 33, 0.25);
    .award-icon {
      height: 72px;
      img {
        max-width: 100%;
        max-height: 100%;
      }
    }
    .award-title {
      @include title;
      color: $brand-primary;

      margin-top: 16px;
      &.award-title__who {
        background: linear-gradient(
          63deg,
          $brand-secondary1-400 27.16%,
          $brand-primary-400 71.03%
        );
        background-clip: text;
        -webkit-background-clip: text;
        -webkit-text-fill-color: transparent;
      }
    }
    .presented-to {
      @include body-regular($font-weight: 400, $font-color: $brand-neutral-700);
      line-height: normal;
      letter-spacing: 0.5px;
      margin-top: 24px;
      text-transform: uppercase;
    }
    .user-name {
      @include title;
      color: $brand-neutral-900;
      font-weight: 400;
      margin-top: 8px;
    }
    .award-description {
      margin-top: 8px;
      @include body-regular($font-weight: 400, $font-color: $brand-neutral-700);
      max-width: 536px;
      text-align: center;
      span {
        font-weight: 700;
      }
    }
    .issued-by {
      padding-top: 24px;
      margin-top: 24px;
      border-top: 1px solid $brand-neutral-200;
      width: 100%;
      @include flex-horizontal-center;
      &.award-issued__by {
        flex-direction: column;
        align-items: center;
        justify-content: center;
        gap: 8px;
        .issued-text {
          align-items: center;
          gap: 8px;
        }
        .logo {
          display: flex;
          gap: 24px;
          margin-left: 0;
          img {
            width: 60px;
            height: 40px;
            object-fit: contain;
          }
        }
      }
      .issued-text {
        @include flex-column-align-right;
        span {
          &:first-child {
            @include label-medium($font-weight: 500);
            color: $brand-neutral-300;
          }
          &:last-child {
            @include body-medium;
            color: $brand-neutral-700;
          }
        }
      }
      .logo {
        margin-left: 24px;
      }
    }
  }
  .more-information {
    border-radius: 8px;
    background-color: $brand-neutral-0;
    box-shadow: 0px 1px 3px 1px rgba(33, 33, 33, 0.25);
    padding: 16px 16px 24px;
    margin-top: 16px;
    .title {
      @include label-large($font-color: $brand-neutral-300, $font-weight: 500);
    }
    .date-information {
      margin-top: 24px;
      display: flex;
      padding-left: 8px;
      .date-box {
        display: flex;
        align-items: center;
        width: 50%;
        .date {
          margin-left: 8px;
          display: flex;
          flex-direction: column;
          span {
            &:first-child {
              @include label-large(
                $font-color: $brand-neutral-700,
                $font-weight: 500
              );
            }
            &:last-child {
              font-weight: 400;
            }
          }
        }
      }
    }
    .other-information {
      margin-top: 24px;
      padding-top: 24px;
      border-top: 1px solid $brand-neutral-200;
    }
  }
  .disclaimer {
    border-radius: 8px;
    background-color: $brand-neutral-0;
    box-shadow: 0px 1px 3px 1px rgba(33, 33, 33, 0.25);
    padding: 16px 16px 24px;
    margin-top: 16px;
    .title {
      @include label-large($font-color: $brand-neutral-900, $font-weight: 500);
    }
    .description {
      @include body-medium;
      color: $brand-neutral-500;
      margin-top: 8px;
    }
  }
}
</style>
