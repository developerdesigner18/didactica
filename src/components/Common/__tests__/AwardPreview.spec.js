import { createLocalVue, mount } from "@vue/test-utils";
import Accordion from "../Accordion.vue";
import AwardPreview from "../AwardPreview.vue";

// create an extended `Vue` constructor
const localVue = createLocalVue();
localVue.component("Accordion", Accordion);

describe("AwardPreview.vue", () => {
  const defaultProps = {
    courseTitle: "Sample Title",
    awardTitle: "Sample Award Title",
    presentedTo: "Sample Presented To",
    userName: "Sample User Name",
    awardDescription: "Sample Award Description",
    issuedByText: "Sample Issued By Text",
    moreInformationText: "Sample More Information",
    issuedOn: "Issued on",
    expiringOn: "Expiring on",
    disclaimerTitle: "Sample Disclaimer Title",
    disclaimer: "Sample Disclaimer",
    infoArr: [
      { title: "Admission criteria", key: "admission_criteria" },
      { title: "Course description", key: "course_description" },
      { title: "Course activities", key: "course_activity" },
      { title: "Course outcomes", key: "course_outcome" },
      { title: "Assessment", key: "course_assessments" },
      { title: "Other information", key: "other_information" }
    ],
    award: {
      content_type: {
        name: "Course Participation Confirmation",
        slug: "course_participation_confirmation"
      },
      credential: {
        expiration_date: "2027-10-11",
        admission_criteria: "<p>asdsad</p>",
        course_description: "<p>assda</p>",
        issuers: "WHO Academy, UNESS",
        course_activity: "<p>asdsad</p>",
        course_outcome: "<p>asdsad</p>",
        issuer_logos: [
          {
            id: 1,
            logo: "https://arodevelopstorage.blob.core.windows.net/openedx-extension-api/dev/issuer-logos/image002.png?se=2024-07-26T15%3A14%3A22Z&sp=r&sv=2021-08-06&sr=b&sig=yPHm%2B/QEZ4Vd30tidX1sgbQ6WLohmqe8jGTw17sL0qY%3D"
          },
          {
            id: 2,
            logo: "https://arodevelopstorage.blob.core.windows.net/openedx-extension-api/dev/issuer-logos/image002.png?se=2024-07-26T15%3A14%3A22Z&sp=r&sv=2021-08-06&sr=b&sig=yPHm%2B/QEZ4Vd30tidX1sgbQ6WLohmqe8jGTw17sL0qY%3D"
          },
          {
            id: 3,
            logo: "https://arodevelopstorage.blob.core.windows.net/openedx-extension-api/dev/issuer-logos/image002.png?se=2024-07-26T15%3A14%3A22Z&sp=r&sv=2021-08-06&sr=b&sig=yPHm%2B/QEZ4Vd30tidX1sgbQ6WLohmqe8jGTw17sL0qY%3D"
          }
        ]
      }
    }
  };

  it("renders when created", () => {
    const wrapper = mount(AwardPreview, {
      localVue,
      propsData: defaultProps
    });
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.element).toMatchSnapshot(); // Snapshot test
  });

  it("renders all props correctly", () => {
    const wrapper = mount(AwardPreview, {
      localVue,
      propsData: defaultProps
    });
    const propKeys = Object.keys(defaultProps);

    propKeys.forEach((key) => {
      expect(wrapper.props(key)).toBe(defaultProps[key]);
    });
    expect(wrapper.element).toMatchSnapshot(); // Snapshot test
  });

  it("renders Accordion component with correct props", () => {
    const wrapper = mount(AwardPreview, {
      localVue,
      propsData: defaultProps
    });
    const accordion = wrapper.findComponent(Accordion);
    expect(accordion.exists()).toBe(true);
    // expect(accordion.props("items")).toBe(defaultProps.infoArr);
    expect(accordion.props("singleOpen")).toBe(false);
    expect(wrapper.element).toMatchSnapshot(); // Snapshot test
  });
});
