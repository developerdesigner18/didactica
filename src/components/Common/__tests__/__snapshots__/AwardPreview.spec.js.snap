// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AwardPreview.vue renders Accordion component with correct props 1`] = `
<div
  class="award__preview--wrapper"
>
  <div
    class="title"
  >
    Sample Title
  </div>
   
  <div
    class="award-information"
  >
    <div
      class="award-icon"
    >
      <img
        alt="Sample Award Title"
        loading="lazy"
        src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/confirmation-of-participation.svg"
      />
    </div>
     
    <div
      class="award-title"
    >
      
      Sample Award Title
    
    </div>
     
    <div
      class="presented-to"
    >
      Sample Presented To
    </div>
     
    <div
      class="user-name"
    >
      Sample User Name
    </div>
     
    <div
      class="award-description"
    >
      
      Sample Award Description
      
      <span>
        “Sample Title”
      </span>
    </div>
     
    <div
      class="issued-by award-issued__by"
    >
      <div
        class="issued-text"
      >
        <span>
          Sample Issued By Text
        </span>
         
        <span>
          WHO Academy, UNESS
        </span>
      </div>
       
      <div
        class="logo"
      >
        <div>
          <img
            alt="logo"
            loading="lazy"
            src="https://arodevelopstorage.blob.core.windows.net/openedx-extension-api/dev/issuer-logos/image002.png?se=2024-07-26T15%3A14%3A22Z&sp=r&sv=2021-08-06&sr=b&sig=yPHm%2B/QEZ4Vd30tidX1sgbQ6WLohmqe8jGTw17sL0qY%3D"
          />
        </div>
        <div>
          <img
            alt="logo"
            loading="lazy"
            src="https://arodevelopstorage.blob.core.windows.net/openedx-extension-api/dev/issuer-logos/image002.png?se=2024-07-26T15%3A14%3A22Z&sp=r&sv=2021-08-06&sr=b&sig=yPHm%2B/QEZ4Vd30tidX1sgbQ6WLohmqe8jGTw17sL0qY%3D"
          />
        </div>
        <div>
          <img
            alt="logo"
            loading="lazy"
            src="https://arodevelopstorage.blob.core.windows.net/openedx-extension-api/dev/issuer-logos/image002.png?se=2024-07-26T15%3A14%3A22Z&sp=r&sv=2021-08-06&sr=b&sig=yPHm%2B/QEZ4Vd30tidX1sgbQ6WLohmqe8jGTw17sL0qY%3D"
          />
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="more-information"
  >
    <div
      class="title"
    >
      Sample More Information
    </div>
     
    <div
      class="date-information"
    >
      <div
        class="date-box issued-date"
      >
        <div
          class="icon"
        >
          <img
            alt="issued on"
            loading="lazy"
            src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/issued-on.svg"
          />
        </div>
         
        <div
          class="date"
        >
          <span>
            Issued On
          </span>
           
          <span>
            Invalid Date
          </span>
        </div>
      </div>
       
      <div
        class="date-box expiring-date"
      >
        <div
          class="icon"
        >
          <img
            alt="expiration date"
            loading="lazy"
            src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/expiring-on.svg"
          />
        </div>
         
        <div
          class="date"
        >
          <span>
            Expiring on
          </span>
           
          <span>
            11/10/2027
          </span>
        </div>
      </div>
    </div>
     
    <div
      class="other-information"
    >
      <div
        class="accordion-wrapper variant1"
      >
        <div
          class="accordion-item"
        >
          <div
            class="accordion-header"
          >
            <img
              loading="lazy"
              src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/arrow.svg"
            />
             
            <span>
              Admission criteria
            </span>
          </div>
           
          <!---->
        </div>
        <div
          class="accordion-item"
        >
          <div
            class="accordion-header"
          >
            <img
              loading="lazy"
              src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/arrow.svg"
            />
             
            <span>
              Course description
            </span>
          </div>
           
          <!---->
        </div>
        <div
          class="accordion-item"
        >
          <div
            class="accordion-header"
          >
            <img
              loading="lazy"
              src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/arrow.svg"
            />
             
            <span>
              Course activities
            </span>
          </div>
           
          <!---->
        </div>
        <div
          class="accordion-item"
        >
          <div
            class="accordion-header"
          >
            <img
              loading="lazy"
              src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/arrow.svg"
            />
             
            <span>
              Course outcomes
            </span>
          </div>
           
          <!---->
        </div>
        <div
          class="accordion-item"
        >
          <div
            class="accordion-header"
          >
            <img
              loading="lazy"
              src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/arrow.svg"
            />
             
            <span>
              Assessment
            </span>
          </div>
           
          <!---->
        </div>
        <div
          class="accordion-item"
        >
          <div
            class="accordion-header"
          >
            <img
              loading="lazy"
              src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/arrow.svg"
            />
             
            <span>
              Other information
            </span>
          </div>
           
          <!---->
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="disclaimer"
  >
    <div
      class="title"
    >
      Sample Disclaimer Title
    </div>
     
    <div
      class="description"
    >
      
      Sample Disclaimer
    
    </div>
  </div>
</div>
`;

exports[`AwardPreview.vue renders all props correctly 1`] = `
<div
  class="award__preview--wrapper"
>
  <div
    class="title"
  >
    Sample Title
  </div>
   
  <div
    class="award-information"
  >
    <div
      class="award-icon"
    >
      <img
        alt="Sample Award Title"
        loading="lazy"
        src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/confirmation-of-participation.svg"
      />
    </div>
     
    <div
      class="award-title"
    >
      
      Sample Award Title
    
    </div>
     
    <div
      class="presented-to"
    >
      Sample Presented To
    </div>
     
    <div
      class="user-name"
    >
      Sample User Name
    </div>
     
    <div
      class="award-description"
    >
      
      Sample Award Description
      
      <span>
        “Sample Title”
      </span>
    </div>
     
    <div
      class="issued-by award-issued__by"
    >
      <div
        class="issued-text"
      >
        <span>
          Sample Issued By Text
        </span>
         
        <span>
          WHO Academy, UNESS
        </span>
      </div>
       
      <div
        class="logo"
      >
        <div>
          <img
            alt="logo"
            loading="lazy"
            src="https://arodevelopstorage.blob.core.windows.net/openedx-extension-api/dev/issuer-logos/image002.png?se=2024-07-26T15%3A14%3A22Z&sp=r&sv=2021-08-06&sr=b&sig=yPHm%2B/QEZ4Vd30tidX1sgbQ6WLohmqe8jGTw17sL0qY%3D"
          />
        </div>
        <div>
          <img
            alt="logo"
            loading="lazy"
            src="https://arodevelopstorage.blob.core.windows.net/openedx-extension-api/dev/issuer-logos/image002.png?se=2024-07-26T15%3A14%3A22Z&sp=r&sv=2021-08-06&sr=b&sig=yPHm%2B/QEZ4Vd30tidX1sgbQ6WLohmqe8jGTw17sL0qY%3D"
          />
        </div>
        <div>
          <img
            alt="logo"
            loading="lazy"
            src="https://arodevelopstorage.blob.core.windows.net/openedx-extension-api/dev/issuer-logos/image002.png?se=2024-07-26T15%3A14%3A22Z&sp=r&sv=2021-08-06&sr=b&sig=yPHm%2B/QEZ4Vd30tidX1sgbQ6WLohmqe8jGTw17sL0qY%3D"
          />
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="more-information"
  >
    <div
      class="title"
    >
      Sample More Information
    </div>
     
    <div
      class="date-information"
    >
      <div
        class="date-box issued-date"
      >
        <div
          class="icon"
        >
          <img
            alt="issued on"
            loading="lazy"
            src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/issued-on.svg"
          />
        </div>
         
        <div
          class="date"
        >
          <span>
            Issued On
          </span>
           
          <span>
            Invalid Date
          </span>
        </div>
      </div>
       
      <div
        class="date-box expiring-date"
      >
        <div
          class="icon"
        >
          <img
            alt="expiration date"
            loading="lazy"
            src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/expiring-on.svg"
          />
        </div>
         
        <div
          class="date"
        >
          <span>
            Expiring on
          </span>
           
          <span>
            11/10/2027
          </span>
        </div>
      </div>
    </div>
     
    <div
      class="other-information"
    >
      <div
        class="accordion-wrapper variant1"
      >
        <div
          class="accordion-item"
        >
          <div
            class="accordion-header"
          >
            <img
              loading="lazy"
              src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/arrow.svg"
            />
             
            <span>
              Admission criteria
            </span>
          </div>
           
          <!---->
        </div>
        <div
          class="accordion-item"
        >
          <div
            class="accordion-header"
          >
            <img
              loading="lazy"
              src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/arrow.svg"
            />
             
            <span>
              Course description
            </span>
          </div>
           
          <!---->
        </div>
        <div
          class="accordion-item"
        >
          <div
            class="accordion-header"
          >
            <img
              loading="lazy"
              src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/arrow.svg"
            />
             
            <span>
              Course activities
            </span>
          </div>
           
          <!---->
        </div>
        <div
          class="accordion-item"
        >
          <div
            class="accordion-header"
          >
            <img
              loading="lazy"
              src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/arrow.svg"
            />
             
            <span>
              Course outcomes
            </span>
          </div>
           
          <!---->
        </div>
        <div
          class="accordion-item"
        >
          <div
            class="accordion-header"
          >
            <img
              loading="lazy"
              src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/arrow.svg"
            />
             
            <span>
              Assessment
            </span>
          </div>
           
          <!---->
        </div>
        <div
          class="accordion-item"
        >
          <div
            class="accordion-header"
          >
            <img
              loading="lazy"
              src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/arrow.svg"
            />
             
            <span>
              Other information
            </span>
          </div>
           
          <!---->
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="disclaimer"
  >
    <div
      class="title"
    >
      Sample Disclaimer Title
    </div>
     
    <div
      class="description"
    >
      
      Sample Disclaimer
    
    </div>
  </div>
</div>
`;

exports[`AwardPreview.vue renders when created 1`] = `
<div
  class="award__preview--wrapper"
>
  <div
    class="title"
  >
    Sample Title
  </div>
   
  <div
    class="award-information"
  >
    <div
      class="award-icon"
    >
      <img
        alt="Sample Award Title"
        loading="lazy"
        src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/confirmation-of-participation.svg"
      />
    </div>
     
    <div
      class="award-title"
    >
      
      Sample Award Title
    
    </div>
     
    <div
      class="presented-to"
    >
      Sample Presented To
    </div>
     
    <div
      class="user-name"
    >
      Sample User Name
    </div>
     
    <div
      class="award-description"
    >
      
      Sample Award Description
      
      <span>
        “Sample Title”
      </span>
    </div>
     
    <div
      class="issued-by award-issued__by"
    >
      <div
        class="issued-text"
      >
        <span>
          Sample Issued By Text
        </span>
         
        <span>
          WHO Academy, UNESS
        </span>
      </div>
       
      <div
        class="logo"
      >
        <div>
          <img
            alt="logo"
            loading="lazy"
            src="https://arodevelopstorage.blob.core.windows.net/openedx-extension-api/dev/issuer-logos/image002.png?se=2024-07-26T15%3A14%3A22Z&sp=r&sv=2021-08-06&sr=b&sig=yPHm%2B/QEZ4Vd30tidX1sgbQ6WLohmqe8jGTw17sL0qY%3D"
          />
        </div>
        <div>
          <img
            alt="logo"
            loading="lazy"
            src="https://arodevelopstorage.blob.core.windows.net/openedx-extension-api/dev/issuer-logos/image002.png?se=2024-07-26T15%3A14%3A22Z&sp=r&sv=2021-08-06&sr=b&sig=yPHm%2B/QEZ4Vd30tidX1sgbQ6WLohmqe8jGTw17sL0qY%3D"
          />
        </div>
        <div>
          <img
            alt="logo"
            loading="lazy"
            src="https://arodevelopstorage.blob.core.windows.net/openedx-extension-api/dev/issuer-logos/image002.png?se=2024-07-26T15%3A14%3A22Z&sp=r&sv=2021-08-06&sr=b&sig=yPHm%2B/QEZ4Vd30tidX1sgbQ6WLohmqe8jGTw17sL0qY%3D"
          />
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="more-information"
  >
    <div
      class="title"
    >
      Sample More Information
    </div>
     
    <div
      class="date-information"
    >
      <div
        class="date-box issued-date"
      >
        <div
          class="icon"
        >
          <img
            alt="issued on"
            loading="lazy"
            src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/issued-on.svg"
          />
        </div>
         
        <div
          class="date"
        >
          <span>
            Issued On
          </span>
           
          <span>
            Invalid Date
          </span>
        </div>
      </div>
       
      <div
        class="date-box expiring-date"
      >
        <div
          class="icon"
        >
          <img
            alt="expiration date"
            loading="lazy"
            src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/expiring-on.svg"
          />
        </div>
         
        <div
          class="date"
        >
          <span>
            Expiring on
          </span>
           
          <span>
            11/10/2027
          </span>
        </div>
      </div>
    </div>
     
    <div
      class="other-information"
    >
      <div
        class="accordion-wrapper variant1"
      >
        <div
          class="accordion-item"
        >
          <div
            class="accordion-header"
          >
            <img
              loading="lazy"
              src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/arrow.svg"
            />
             
            <span>
              Admission criteria
            </span>
          </div>
           
          <!---->
        </div>
        <div
          class="accordion-item"
        >
          <div
            class="accordion-header"
          >
            <img
              loading="lazy"
              src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/arrow.svg"
            />
             
            <span>
              Course description
            </span>
          </div>
           
          <!---->
        </div>
        <div
          class="accordion-item"
        >
          <div
            class="accordion-header"
          >
            <img
              loading="lazy"
              src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/arrow.svg"
            />
             
            <span>
              Course activities
            </span>
          </div>
           
          <!---->
        </div>
        <div
          class="accordion-item"
        >
          <div
            class="accordion-header"
          >
            <img
              loading="lazy"
              src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/arrow.svg"
            />
             
            <span>
              Course outcomes
            </span>
          </div>
           
          <!---->
        </div>
        <div
          class="accordion-item"
        >
          <div
            class="accordion-header"
          >
            <img
              loading="lazy"
              src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/arrow.svg"
            />
             
            <span>
              Assessment
            </span>
          </div>
           
          <!---->
        </div>
        <div
          class="accordion-item"
        >
          <div
            class="accordion-header"
          >
            <img
              loading="lazy"
              src="https://whoalxppublicstorage.blob.core.windows.net/didactica/assets/images/awards/arrow.svg"
            />
             
            <span>
              Other information
            </span>
          </div>
           
          <!---->
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="disclaimer"
  >
    <div
      class="title"
    >
      Sample Disclaimer Title
    </div>
     
    <div
      class="description"
    >
      
      Sample Disclaimer
    
    </div>
  </div>
</div>
`;
